### PR TITLE
Make syncing with s3 always reset artifacts to only cur / prev

### PR DESCRIPTION
**Pull request by anlu**
There were two situations that caused leftover artifacts in the old FTL system.

1. Artifacts that were older than package.previous on s3 were completely disregarded on the localhosts, and would never be removed.
2. Any artifacts built that were not deployed would also get downloaded to individual hosts.

This change forces ONLY the artifacts labeled package.current and package.previous in s3 to be copied to a host when ftl sync is run. All other artifacts are purged. This saves us from the disk space warnings forcing manual cleanup.

This does make ftl jump on a local machine useless, but people should not be doing that anyways. ftl jump-back still works as before, although this should not be run on an individual local host either. Essentially, the state in s3 is the canonical state of artifacts, and all hosts will always reflect that state on ftl sync.

**Deploy Plan**
Moving the ftl binary to each host that requires it should do it. I'd like to do this carefully, with stage hosts first to make sure everything is working as expected. How to build ftl lol?

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -192,6 +145,32 @@ func downloadRemoteRevisions(r *ftl.RemoteRepository, l *ftl.LocalRepository, re
 	return nil
 }
 
+func containsRevision(revList []*ftl.RevisionInfo, info *ftl.RevisionInfo) bool {
```
Comment (bickfordb) [inline main.go]: Add godoc

File: main_test.go [hunk 2]
```diff
@@ -5,193 +5,72 @@ import (
...
 )
 
-func Test_syncPackage_downNone(t *testing.T) {
-	rr := []*ftl.RevisionInfo{
-		{"test", "001"},
-		{"test", "002"},
-		{"test", "003"},
-	}
-
+func Test_syncPackage_sameState(t *testing.T) {
```
Comment (bickfordb) [inline main_test.go]: Rename this to `TestSyncPackageSameState` and add a GoDoc describing it

File: main_test.go [hunk 3]
```diff
@@ -5,193 +5,72 @@ import (
...
+		t.Error("Expected 1 purge, got", len(purge))
 	}
 }
 
-func Test_syncPackage_purgeOne(t *testing.T) {
-	rr := []*ftl.RevisionInfo{
-		{"test", "002"},
-	}
-
+func Test_syncPackage_updateOne(t *testing.T) {
```
Comment (bickfordb) [inline main_test.go]: Same as above

File: main_test.go [hunk 4]
```diff
@@ -5,193 +5,72 @@ import (
...
-		t.Error("Expected purge 001")
+		t.Error("Expected 1 purge, got", len(purge))
 	}
 }
 
-func Test_syncPackage_purgeAll(t *testing.T) {
-	rr := []*ftl.RevisionInfo{
-	}
 
+func Test_syncPackage_updateTwo(t *testing.T) {
```
Comment (bickfordb) [inline main_test.go]: Fix name, add Go Doc

File: main.go [hunk 5]
```diff
@@ -84,41 +84,7 @@ func removePackageRevision(local *ftl.LocalRepository, revision *ftl.RevisionInf
...
-			purgeRevs = append(purgeRevs, localRevisions[localNdx])
-			localNdx++
-		}
-	}
-
-	return
-}
-
-func retrieveRemoteRevisions(r *ftl.RemoteRepository, packageName string) (curRev, prevRev *ftl.RevisionInfo, revisions []*ftl.RevisionInfo, err error) {
+func retrieveRemoteRevisions(r *ftl.RemoteRepository, packageName string) (curRev, prevRev *ftl.RevisionInfo, err error) {
```
Comment (bickfordb) [inline main.go]: Add a Go Doc describing this function while it's fresh in your head

**Comment by bickfordb** [comment]
:+1: after you add some documentation, fix the odd camel-snake-case function names

**Comment by anlu** [comment]
Documentation added--also got rid of RevisionListResult because I realized we don't need it anymore. Won't actually show up in godoc though, since they're private.

**Comment by bickfordb** [comment]
:+1:

**Pull request closed without merging**